### PR TITLE
Mhv 59706 blocked tg alert fix

### PR DESCRIPTION
**Pull request by KyleCardwell**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

## Summary

- bug:  on an edit draft, if the page is refreshed, the blocked triage group alert would occasionally appear, even though there were no blocked triage groups.  This was happening because the Edit draft page would load before the recipients api call would finish.
fix: adding a condition to the useEffect where "isLoaded" is set to check whether or not the values in the recipients slice of redux have been set, and only loading the draft it it is not "undefined"

### [MHV-59706](https://jira.devops.va.gov/browse/MHV-59706) - Blocked TG warning message in mobile and desktop view, even when user is not blocked from TG. ###

> User see the blocked TG waring message in mobile AND DESKTOP view, even when user is not blocked from TG. This happening only in the Draft folder.
> 
> TG Alert identifier: "blocked-triage-group-alert"
> 
> Step to Reproduce.
> 1) Login to va.gov SM
> 2) Navigate to Drafts folder
> 3) Open any draft message.
> 4) Change web view from mobile view
> 5) Refresh the page
> 6) User see the warning message "Your account is no longer connected to TG "
> 
> Feature Flag Y/N ? N
> 
> Manual Testing Y/N ? Y-Rakesh
> 
> Automated Testing Y/N ? N
> 
> Accessibility Testing Y/N ? N
> 
> UCD Validation Y/N N
> 
> Acceptance Criteria:
> 
> AC1 User does not see the blocked triage group warning message in the scenario described above

## Testing done

- manual testing, updated unit test

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

va.gov - secure messaging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/containers/ThreadDetails.jsx [hunk 1]
```diff
@@ -147,7 +147,11 @@ const ThreadDetails = props => {
         </>
       );
     }
-    if (drafts?.length === 1 && !messages?.length) {
+    if (
+      drafts?.length === 1 &&
+      !messages?.length
+      // recipients.allowedRecipients.length > 0
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/containers/ThreadDetails.jsx:153]: commented code

File: src/applications/mhv-secure-messaging/tests/containers/ThreadDetails.unit.spec.jsx [hunk 2]
```diff
@@ -760,4 +761,53 @@ describe('Thread Details container', () => {
...
+      messagesActions,
+      'retrieveMessageThread',
+    );
+
+    setup(state);
+
+    await waitFor(() => {
+      expect(retrieveMessageThreadSpy.calledOnce).not.to.be.true;
+      retrieveMessageThreadSpy.restore();
+    });
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/tests/containers/ThreadDetails.unit.spec.jsx:811]: this test may need to be revised. I was trying to break it by adding those redux values and it was still passing

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
@KyleCardwell I am glad we are fixing this annoying defect 🥳 Below I pointed the commented code and the unit test may need to be revised

**Review by alex-morgun-1 — APPROVED** [review]
this looks much better. Tested locally and works as expected. Good unit test coverage

**Pull request merged**